pykechain/client.py

```diff
@@ -1105,6 +1105,106 @@ def create_activity(
             parent._cached_children.append(new_activity)
         return new_activity
 
+    def clone_activities(
+        self,
+        activities: List[Union[Activity2, Text]],
+        parent: Union[Activity2, Text],
+        activity_update_dicts: Optional[Dict] = None,
+        clone_parts: Optional[bool] = False,
+        clone_part_instances: Optional[bool] = True,
+        clone_children: Optional[bool] = True,
+        excluded_parts: Optional[List[Text]] = None,
+        part_parent_model: Optional[Union[Part2, Text]] = None,
+        part_parent_instance: Optional[Union[Part2, Text]] = None,
+        part_model_rename_template: Optional[Text] = None,
+        part_instance_rename_template: Optional[Text] = None,
+        asynchronous: Optional[bool] = False,
+        **kwargs
+    ) -> List[Activity2]:
+        """
+        Clone multiple activities.
+
+        :param activities: list of Activity2 object or UUIDs
+        :type activities: list
+        :param parent: parent Activity2 sub-process object or UUID
+        :type parent: Activity2
+        :param activity_update_dicts: (O) dict of dictionaries, each key-value combination relating to an activity
+        to clone and a dict of new values to assign, e.g. `{activity.id: {"name": "Cloned activity"}}`
+        :type activity_update_dicts: dict
+        :param clone_parts: (O) whether to clone the data models configured in the activities, defaults to False
+        :type clone_parts: bool
+        :param clone_part_instances: (O) whether to clone the part instances of the data model configured in the
+            activities, defaults to True
+        :type clone_part_instances: bool
+        :param clone_children: (O) whether to clone child parts
+        :type clone_children: bool
+        :param excluded_parts: (O) list of Part2 objects or UUIDs to exclude from being cloned,
+            maintaining the original configuration of the widgets.
+        :type excluded_parts: list
+        :param part_parent_model: (O) parent Part2 object or UUID for the copied data model(s)
+        :type part_parent_model: Part2
+        :param part_parent_instance: (O) parent Part2 object or UUID for the copied part instance(s)
+        :type part_parent_instance: Part2
+        :param part_model_rename_template: (O) renaming template for part models. Must contain "{name}"
+        :type part_model_rename_template: str
+        :param part_instance_rename_template: (O) renaming template for part instances. Must contain "{name}"
+        :type part_instance_rename_template: str
+        :param asynchronous: If true, immediately returns without activities (default = False)
+        :type asynchronous: bool
+        :return: list of cloned activities
+        :rtype: list
+        :raises APIError if cloned
+        """
+        update_name = 'activity_update_dicts'
```

Comment:
you might want to consider checking the version of the backend to see if the API is in place.
The `wim` version with the bulk_clone_activities with parts is version `>= 3.6.0`